src/libraries/System.Drawing.Common/src/System/Drawing/Gdiplus.cs

```diff
@@ -18,13 +15,19 @@ internal static partial class SafeNativeMethods
         internal static partial class Gdip
         {
             private static readonly IntPtr s_initToken;
-            private const string ThreadDataSlotName = "system.drawing.threaddata";
+
+            [ThreadStatic]
+            private static IDictionary? t_threadData;
 
             static Gdip()
             {
-                Debug.Assert(s_initToken == IntPtr.Zero, "GdiplusInitialization: Initialize should not be called more than once in the same domain!");
+                if (!OperatingSystem.IsWindows())
+                {
+                    NativeLibrary.SetDllImportResolver(Assembly.GetExecutingAssembly(), static (_, _, _) =>
+                        throw new PlatformNotSupportedException(SR.PlatformNotSupported_Unix));
+                }
```

Comment:
I didn't follow on the official announcement but I assume it's intentionally that we explicitly block customers from loading this assembly on Unix?

Reply:
That was the original plan. I first wrote

``` csharp
if (!OperatingSystem.IsWindows())
{
    throw new PlatformNotSupportedException(SR.PlatformNotSupported_Unix);
}
```

on the static constructor, which would throw more aggressively on non-Windows, but [after feedback](https://github.com/dotnet/runtime/pull/64623#pullrequestreview-869971017) I made it throw only if a P/Invoke from the SDC assembly was going to be called, which is the actual thing that is not supported on Unix, in the strictest sense. I don't think that just loading the assembly on Unix will cause any failures.